pkg/commands/git_commands/github.go

```diff
@@ -0,0 +1,436 @@
+package git_commands
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cli/go-gh/v2/pkg/auth"
+	gogit "github.com/jesseduffield/go-git/v5"
+	"github.com/jesseduffield/lazygit/pkg/commands/models"
+	"github.com/samber/lo"
+	"golang.org/x/sync/errgroup"
+)
+
+type GitHubCommands struct {
+	*GitCommon
+}
+
+func NewGitHubCommand(gitCommon *GitCommon) *GitHubCommands {
+	return &GitHubCommands{
+		GitCommon: gitCommon,
+	}
+}
+
+// https://github.com/cli/cli/issues/2300
+func (self *GitHubCommands) BaseRepo() error {
+	cmdArgs := NewGitCmd("config").
+		Arg("--local", "--get-regexp", ".gh-resolved").
+		ToArgv()
+
+	return self.cmd.New(cmdArgs).DontLog().Run()
+}
+
+// Ex: git config --local --add "remote.origin.gh-resolved" "jesseduffield/lazygit"
+func (self *GitHubCommands) SetBaseRepo(repository string) (string, error) {
+	cmdArgs := NewGitCmd("config").
+		Arg("--local", "--add", "remote.origin.gh-resolved", repository).
+		ToArgv()
+
+	return self.cmd.New(cmdArgs).DontLog().RunWithOutput()
+}
+
+type Response struct {
+	Data RepositoryQuery `json:"data"`
+}
+
+type RepositoryQuery struct {
+	Repository map[string]PullRequest `json:"repository"`
+}
+
+type PullRequest struct {
+	Edges []PullRequestEdge `json:"edges"`
+}
+
+type PullRequestEdge struct {
+	Node PullRequestNode `json:"node"`
+}
+
+type PullRequestNode struct {
+	Title               string                `json:"title"`
+	HeadRefName         string                `json:"headRefName"`
+	Number              int                   `json:"number"`
+	Url                 string                `json:"url"`
+	HeadRepositoryOwner GithubRepositoryOwner `json:"headRepositoryOwner"`
+	State               string                `json:"state"`
+}
+
+type GithubRepositoryOwner struct {
+	Login string `json:"login"`
+}
+
+func fetchPullRequestsQuery(branches []string, owner string, repo string) string {
+	var queries []string
+	for i, branch := range branches {
```

Comment:
You can do this in one request (not sure about the length limits for the query) with:

```graphql
query {
  search(first:3, query:"head:gh-integration-3 head:integration-tests-on-windows repo:jesseduffield/lazygit", type: ISSUE) {
    nodes {
      ... on PullRequest {
        url
      }
    }
  }
}
```